client/src/js/addendum/AddendumForm.jsx

```diff
@@ -0,0 +1,284 @@
+import React, { useState } from 'react'
+import PropTypes from 'prop-types'
+import { makeStyles } from '@material-ui/core/styles'
+import { Addendum, AddendumType } from '../../common/model/addendum'
+import {
+  Box,
+  Button,
+  Card,
+  Grid
+} from '@material-ui/core'
+import { Agreement, AgreementType } from '../../common/model/agreement'
+import IdentityForm from '../identity/IdentityForm'
+import IdentityCard from './IdentityCard'
+import * as _ from 'lodash'
+import { Alert } from '@material-ui/lab'
+import { Identity, IdentityType } from '../../common/model/identity'
+import { Whitelist } from '../../common/model/whitelists'
+
+const useStyles = makeStyles(theme => ({
+  root: {
+    padding: theme.spacing(2),
+    marginTop: theme.spacing(2)
+  },
+  updateFormContainer: {
+    width: '100%'
+  },
+  bold: {
+    fontWeight: 'bold'
+  }
+}))
+
+/**
+ * This class contains the logic for the AddendumForm component
+ */
+class AddendumFormCtrl {
+  /**
+   * Check if a User owns this agreement. For now this is true only when the User is the Signer of the Agreement.
+   * This may change when we support co-signers/managers.
+   * @param user {{email: string}} The Firebase user object
+   * @param agreement {Agreement} An Agreement object
+   * @returns {Promise<boolean>}
+   */
+  static async isOwnerOrManager (user, agreement) {
```

Comment:
nit: consistent temrinology
```suggestion
  static async isSignerOrManager (user, agreement) {
```